src/App.js

```diff
@@ -7,54 +7,59 @@ import "./App.css";
 // Save the Firebase message folder name as a constant to avoid bugs due to misspelling
 const MESSAGE_FOLDER_NAME = "messages";
 
-class App extends React.Component {
-  constructor(props) {
-    super(props);
-    // Initialise empty messages array in state to keep local state in sync with Firebase
-    // When Firebase changes, update local state, which will update local UI
-    this.state = {
-      messages: [],
-    };
-  }
-
-  componentDidMount() {
-    const messagesRef = ref(database, MESSAGE_FOLDER_NAME);
+const App = () => {
+  const [messages, setMessages] = useState([]);
+  const [input, setInput] = useState("");
+  const messagesRef = ref(database, MESSAGE_FOLDER_NAME);
+  useEffect(() => {
     // onChildAdded will return data for every child at the reference and every subsequent new child
+    const msg = [];
     onChildAdded(messagesRef, (data) => {
+      msg.push({ key: data.key, val: data.val() });
       // Add the subsequent child to local component state, initialising a new array to trigger re-render
-      this.setState((state) => ({
-        // Store message key so we can use it as a key in our list items when rendering messages
-        messages: [...state.messages, { key: data.key, val: data.val() }],
-      }));
+      setMessages([...msg]);
     });
-  }
-
+  }, []);
   // Note use of array fields syntax to avoid having to manually bind this method to the class
-  writeData = () => {
+
+  const messageListItems = messages.map((message) => (
+    <li key={message.key}>{message.val}</li>
+  ));
+
+  const handleChange = (e) => {
+    setInput(e.target.value);
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
     const messageListRef = ref(database, MESSAGE_FOLDER_NAME);
     const newMessageRef = push(messageListRef);
-    set(newMessageRef, "abc");
+    set(newMessageRef, input);
+    setInput("");
   };
 
-  render() {
-    // Convert messages in state to message JSX elements to render
-    let messageListItems = this.state.messages.map((message) => (
-      <li key={message.key}>{message.val}</li>
-    ));
-    return (
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <p>
-            Edit <code>src/App.js</code> and save to reload.
-          </p>
-          {/* TODO: Add input field and add text input as messages in Firebase */}
-          <button onClick={this.writeData}>Send</button>
-          <ol>{messageListItems}</ol>
-        </header>
-      </div>
-    );
-  }
-}
+  return (
+    <div className="App">
+      <header className="App-header">
+        <img src={logo} className="App-logo" alt="logo" />
+        <br></br>
+        <form onSubmit={handleSubmit}>
+          <label for="message">
+            To start communicating, type a message inside the input field!
+          </label>
+          <br></br>
```

Comment:
br can be self-closing. We can also use css margins for the same purpose, which is usually rather desired